### ISSUE TITLE
Bug: `no-else-after-return` does not allow `else if`'s by default like ESLint's `no-else-return` does

### ISSUE THREAD

**Issue by tbrisbane** [ajafff/tslint-consistent-codestyle#43]
### Issue type
Bug
### Version
`v1.10.0`
### Overview
The `no-else-after-return` rule seems to, by default, be issuing warnings when `else if`'s are used after `return` statements.

The [documentation for `no-else-after-return`](https://github.com/ajafff/tslint-consistent-codestyle/blob/master/docs/no-else-after-return.md) states that it "Works like [no-else-return from eslint](http://eslint.org/docs/rules/no-else-return)", and (at least for the current version of ESLint) the default behaviour of `no-else-return` allows `return` statements after `else if`'s.
### Current behaviour
When `no-else-after-return: true` is specified, the following code causes an `unnecessary else after return` warning to appear for me:
```js
const val = Math.random();

if (val > 0.5) {
  return 1;
} else if (val < 0.5) {
  return -1;
}

return 0;
```
### Expected behaviour
When `no-else-after-return: true` is specified, the above code should *not* issue a warning.

If `no-else-after-return: [true, { "allowElseIf": false }]` was specified, then the above code *should* issue a warning.
### Additional context
I use this rule as part of [tslint-config-airbnb](https://github.com/progre/tslint-config-airbnb/), which added `no-else-after-return` to their ruleset in `v5.4.1`.

**Comment by ajafff** [comment]
I agree that the rules doesn't offer what users expect from reading the docs.
Though I'm not sure if I really want to fix this. `no-else-after-return` is superseded by `no-unnecessary-else`. I'm inclined to rather deprecate the rule.
Until then, the docs should be updated.

**Comment by mummybot** [comment]
You now have a very popular TSLint library which is pulled in by [AirBnB](https://github.com/progre/tslint-config-airbnb/blob/master/tslint.js#L5), which means the "whole world" is getting it :)

We would very much like the inability to set the arguments as per eslint for this rule, as we don't want to use the ```no-unecessary-else```.

**Comment by ajafff** [comment]
I don't understand why you don't want to use `no-unnecessary-else`, but that's none of my business.

I actually wanted to deprecate and eventually remove `no-else-after-return` because it contained a lot of duplicated and slightly modified code to handle *only* return statements as control flow end. The rule also had quite a few bugs that never got fixed.
Fortunately I have refactored my control flow analysis and can reuse it for this rule. There's no more duplicate code and correct control flow analysis.

### TL:DR

I'm going to update the rule and also add the requested option.

But there's one thing I don't understand: ESLint's docs state that the following code is invalid even if `allowElseIf` is enabled. Why?

```js
function foo() {
    if (x) {
        return y;
    } else if (z) {
        return w;
    } else {
        return t;
    }
}
```

**Comment by mummybot** [comment]
Thanks, you're a star! As for your question re the code being invalid (if I get the problem):

```javascript
function foo() {
    if (x) {
        return y;
    } else if (z) {
        return w;
    } else { // this last else is unnecessary, as it will just fall through
        return t;
    }
}
```
e.g.
```javascript
function foo() {
    if (x) {
        return y;
    } else if (z) {
        return w;
    }
    return t;
}
```

**Comment by ajafff** [comment]
@mummybot That makes sense, thanks.

The PR for this option is up at #61.
I'll release this tomorrow.

**Comment by ajafff** [comment]
The new option is published in v1.13.0
